app/service/planning_svc.py

```diff
@@ -157,8 +157,19 @@ async def get_links(self, operation, buckets=None, agent=None, trim=True):
         if agent:
             links.extend(await self.generate_and_trim_links(agent, operation, abilities, trim))
         else:
+            pre_filter = []
+            parallel_list = []
             for agent in operation.agents:
-                links.extend(await self.generate_and_trim_links(agent, operation, abilities, trim))
+                pre_filter.append(await self.generate_and_trim_links(agent, operation, abilities, trim))
+            for agent_list in pre_filter:
+                for el in agent_list:
+                    if not self._fil(el, 'remote.host.fqdn'):
```

Comment:
So the logic here is:

    if command doesnt use 'remote.host.fqdn', no issue, add to links returned.

    elif command does use 'remote.host.fqdn', then only add if it's unique across all agents(?)

--> So issue was a planner was grabbing multiple duplicates of commands with 'remote.host.fqdn' in one call to get_links() and then pushing those links all onto operation?

Reply:
Yep - each agent would generate it's own version of the command, and because of the way we were computing actions across the agent space, we'd end up not accounting for that, and submitting them.